taskscheduler/scheduler_test.go

```diff
@@ -0,0 +1,167 @@
+package taskscheduler
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestScheduler(t *testing.T) {
+	s, err := NewScheduler(5, 10)
+	if err != nil {
+		t.Error(err)
```

Comment:
```suggestion
		t.Fatalf("%v", err)
```

Let's use `t.Fatalf` when we're detecting an issue that should immediately fail and terminate the test.

Reply:
Done.